### ISSUE TITLE
Should angular-translate support case insensitive interpolated variables?

### ISSUE THREAD

**Issue by vikeen** [angular-translate/angular-translate#1520]
### Should angular-translate support case insensitive interpolated variables?
My company uses a translating service that returned this translation value to us in simplified chinese.

```json
"percentApr": "{{NUM}}％年利率 (APR)",
```

The bound angular variable `{{NUM}}` was unintentionally converted to uppercase and this prevents the translate service from picking up the value in our view

```html
<li translate="maker.percentApr" translate-value-num="{{MarketStatsBarCtrl.stats.availableAprWeightedAvg}}"></li>
```

### Your environment
* angular-translate: 2.7.2
* angular: 1.3.16
* Browser: Google Chrome version 51.0.2704.103 (64-bit)

### Steps to reproduce
Demo: https://plnkr.co/edit/ig32Ui?p=preview

You can see from the demo that the variables in the translate are switched to uppercase versions and no longer get bound in during runtime.

### Expected behaviour
I didn't expect this to actually cause an issue because I don't see a reason why it would matter if a translation key parameter was `num`, `NUM`, or `Num`.

### Actual behaviour
The translate variable `num` is never found because our system passed in `{{NUM}}` instead.

I believe that the translate keys have no reason being case insensitive. I realize that this not angular-translate's fault, but I can't think of a reason why it would matter if the translation key parameters were case insensitive.

**Comment by knalli** [comment]
Internally, we use the interpolation service which actually relies on simple JavaScript. An object property `num` is not the same as `NUM`.

```js
let obj = {};
obj.num = 1;
obj.NUM = 2;

assert(obj.num === 1);
assert(obj.NUM === 2);
```

tl;dr: I'm answering your question with: No, it should not.

**Comment by vikeen** [comment]
@knalli, fair enough. Thank you for clarifying and acknowledging this.
